src/server/db_slice.cc

```diff
@@ -328,25 +328,32 @@ void DbSlice::AutoUpdater::Run() {
   if (fields_.action == DestructorAction::kDoNothing) {
     return;
   }
-  // TBD add logic to update iterator if needed as we can now preempt in cb
 
   // Check that AutoUpdater does not run after a key was removed.
   // If this CHECK() failed for you, it probably means that you deleted a key while having an auto
   // updater in scope. You'll probably want to call Run() (or Cancel() - but be careful).
   DCHECK(IsValid(fields_.db_slice->db_arr_[fields_.db_ind]->prime.Find(fields_.key)))
       << "Key was removed before PostUpdate() - this is a bug!";
 
-  // Make sure that the DB has not changed in size since this object was created.
-  // Adding or removing elements from the DB may invalidate iterators.
-  CHECK_EQ(fields_.db_size, fields_.db_slice->DbSize(fields_.db_ind))
-      << "Attempting to run post-update after DB was modified";
-
-  CHECK_EQ(fields_.deletion_count, fields_.db_slice->deletion_count_)
-      << "Attempting to run post-update after a deletion was issued";
-
   DCHECK(fields_.action == DestructorAction::kRun);
   CHECK_NE(fields_.db_slice, nullptr);
 
+  if (EngineShardSet::IsTieringEnabled()) {
+    // When triering is enabled we can preempt on write to disk, therefor it can be invalidated
+    // untill we run the post updated.
+    if (!fields_.it.IsOccupied() || fields_.it->first != fields_.key) {
```

Comment:
this check appears in several places now. Consider introducing an utility function:
`DbTable::Launder(PrimeIterator it, string_view key) -> PrimeIterator` that does that

Reply:
Do you not plan to do this?

Reply:
just added. Though I dont understand the function name suggested.. can you explain

Reply:
I think Vlad came up with that name, I'm not sure why launder specifically, but it sounds cool so I don't argue :sunglasses: